### PR TITLE
Implement elle core

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve? 

Implement elle's core module.

### Check List 

Tests 

 - Unit test

Code changes

 - Has Go code change

### CODE REVIEW

File: pkg/elle/core/core.go [hunk 1]
```diff
@@ -54,22 +79,142 @@ func (c *CycleExplainer) renderCycleExplanation(explainer DataExplainer, circle
 	panic("impl me")
 }
 
-//  A function which takes a history and returns a {:graph, :explainer, :anomalies} map; e.g. realtime-graph.
-type Analyzer func(history History) (Anomalies, DirectedGraph, DataExplainer)
+// Analyzer is a function which takes a history and returns a {:graph, :explainer} map; e.g. realtime-graph.
+type Analyzer func(history History) (DirectedGraph, DataExplainer)
```
Comment (mahjonp) [inline pkg/elle/core/core.go:83]: Perhaps we need retain `Anomalies`, it's used on `Combine` func and rw_register workload.

File: pkg/elle/core/core.go [hunk 2]
```diff
@@ -54,22 +79,142 @@ func (c *CycleExplainer) renderCycleExplanation(explainer DataExplainer, circle
...
-	panic("implement me")
+func ProcessGraph(history History) (DirectedGraph, DataExplainer) {
+	var (
+		okHistory = history.FilterType(OpTypeOk)
+		processes map[int]struct{}
+		graphs    []DirectedGraph
+	)
+
+	for _, op := range okHistory {
+		if op.Process != 0 {
```
Comment (mahjonp) [inline pkg/elle/core/core.go:132]: Why filter out Process 0? what's Process 0 means?

Comment (you06) [reply pkg/elle/core/core.go:132]: Process should not be 0 and 0 will be the default value when `Op` struct inits. Filter out Process with value 0 to exclude the operations which don't have a Process value.

Comment (mahjonp) [reply pkg/elle/core/core.go:132]: How about change Process type to *int?

**Pull request merged**